lib/syskit/test/profile_assertions.rb

```diff
@@ -567,6 +602,37 @@ def validate_actions(action_or_profile, exclude: [])
 
                 actions
             end
+
+            # @api private
+            # Assert actions, flunking in case actions could not be validated, excluding
+            # those included in 'exclude'.
+            #
+            # @param action_or_profile an action interface or profile
+            def assert_actions(action_or_profile, exclude: [])
+                validate_actions(action_or_profile, exclude: exclude) do |skip|
+                    caller_method = caller[2].split("`").last.split("'").first
+                    flunk "could not validate some non-Syskit actions: #{skip}, " \
+                          "probably because of required arguments. Pass the action to " \
+                          "the 'exclude' option of #{caller_method}, and add a separate " \
+                          "assertion test with the arguments added explicitly"
+                end
+            end
+
+            # @api private
+            # Assert together_with actions, flunking in case actions could not be
+            # validated,excluding those included in 'exclude'.
+            #
+            # @param together_with an action interface or profile
+            def assert_together_with(together_with, exclude: [])
+                validate_actions(together_with, exclude: exclude) do |skip|
+                    caller_method = caller[2].split("`").last.split("'").first
```

Comment:
Ruby 2.7+ has `caller_locations` which returns objects and removes the need for this.